scripts/doiScript.js

```diff
@@ -0,0 +1,83 @@
+var links = document.getElementsByTagName('a');
+var doi=[];
+var count = 0;
+var wbCount = 0;
+for(var i=0;i<links.length;i++){
+    if(links[i].getAttribute('href') && links[i].getAttribute('href').indexOf('doi')>=0){
+       doi.push(links[i].getAttribute('href').slice(10));
+    }
+}
+var result=[];
+var wbResult = [];
+var length = doi.length;
+function counterInc(){
+    count++;
+    if(count== length){
+        requestForAll();
+    }
+}
+function requestForAll(){
+    for(var i=0;i<result.length;i++){
+        if(result.url!=null){
+            var wbXhr = new XMLHttpRequest();
+            wbLinkFetch(wbXhr,result[i]);
+        }
+    }
+}
+function wbcounterInc(){
+    wbCount++;
+    if(wbCount== result.length){
+        chrome.runtime.sendMessage({doi:wbResult,src:src,show:show,remove:remove,update:update});
+    }
+}
+function wbLinkFetch(wbXhr,result){
+    wbXhr.open('GET','https://archive.org/wayback/available?url='+result.url,true);
+    wbXhr.onerror= function(){
+        wbcounterInc();
+    };
+    try{
+       wbXhr.onload = function(){
+        if(wbXhr.status!=200){
+            wbcounterInc();
+        }else{
+           var wbResponse = JSON.parse(wbXhr.responseText);
+        if(wbResponse.archived_snapshots.closest){
+            var link = wbResponse.archived_snapshots.closest.url;
+            wbResult.push({doi:result.doi,url:link,title:result.title});
+            wbcounterInc();
+        }else{
+            wbResult.push({doi:result.doi,url:null,title:result.title});
+            wbcounterInc();
+        } 
+        }
+        
+    }; 
+    }catch(e){
+        wbcounterInc();
+    }
+    
+    wbXhr.send();
+}
+function checkAvailability(doi,xhttp,result){
+    xhttp.open('GET','https://scholar.archivelab.org/id/doi/'+doi,true);
```

Comment:
Could you make the WayBack URL of the paper available too ? If you see https://scholar.archivelab.org/id/doi/10.1260/0958305054672385 the `url` property of a paper points to the actual paper
@markjohngraham wants the user to be able to see the paper on a web.archive.org URL. I am using the availability API to get that link but it would be great if the call to `https://wwwb-api.archive.org/services/context/papers?url=XXX` returns the WayBack URL too

Reply:
I get it. I will do that.

Reply:
Thank you 
I will make the changes to the code accordingly to the data returned by `https://wwwb-api.archive.org/services/context/papers?url=XXX`